pkg/storage/etcd.go

```diff
@@ -36,37 +37,75 @@ func NewEtcdStorage(endpoints []string) (*EtcdStorage, error) {
 }
 
 // Store stores the key
-func (etcds *EtcdStorage) Store(key string, value Data) error {
+func (etcds *EtcdStorage) StoreCheckin(key string, value time.Time) error {
 	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
```

Comment:
context.Background() should called in the codebase only once to respond to cancellations throughout the system. But, since it was that way before, it's fine in this PR.